arangod/IResearch/IResearchLinkCoordinator.cpp

```diff
@@ -129,110 +129,38 @@ IResearchLinkCoordinator::IResearchLinkCoordinator(
   _sparse = true;  // always sparse
 }
 
-bool IResearchLinkCoordinator::operator==(LogicalView const& view) const noexcept {
-  return _view && _view->id() == view.id();
-}
-
-arangodb::Result IResearchLinkCoordinator::init(
-    arangodb::velocypack::Slice const& definition
-) {
-  std::string error;
-
-  if (!_meta.init(definition, error)) {
-    return arangodb::Result(
-      TRI_ERROR_BAD_PARAMETER,
-      std::string("error parsing view link parameters from json: ") + error
-    );
-  }
-
-  if (!definition.isObject()
-      || !definition.get(StaticStrings::ViewIdField).isString()) {
-    return arangodb::Result(
-      TRI_ERROR_ARANGO_DATA_SOURCE_NOT_FOUND,
-      std::string("error finding view for link '") + std::to_string(Index::id()) + "'"
-    );
-  }
-
-  auto idSlice = definition.get(StaticStrings::ViewIdField);
-  auto viewId = idSlice.copyString();
-  auto& vocbase = Index::collection().vocbase();
-
-  TRI_ASSERT(ClusterInfo::instance());
-  auto logicalView  = ClusterInfo::instance()->getView(vocbase.name(), viewId);
-
-  if (!logicalView
-      || arangodb::iresearch::DATA_SOURCE_TYPE != logicalView->type()) {
-    return arangodb::Result(
-      TRI_ERROR_ARANGO_DATA_SOURCE_NOT_FOUND,
-      std::string("error looking up view '") + viewId + "': no such view"
-    );
-  }
-
-#ifdef ARANGODB_ENABLE_MAINTAINER_MODE
-  auto view = std::dynamic_pointer_cast<IResearchViewCoordinator>(logicalView);
-#else
-  auto view = std::static_pointer_cast<IResearchViewCoordinator>(logicalView);
-#endif
-
-  if (!view) {
-    return arangodb::Result(
-      TRI_ERROR_ARANGO_DATA_SOURCE_NOT_FOUND,
-      std::string("error finding view '") + viewId + "' for link '" + std::to_string(Index::id()) + "'"
-    );
-  }
-
-  _view = view;
-
-  return IResearchLink::init(definition);
-}
-
-bool IResearchLinkCoordinator::matchesDefinition(VPackSlice const& slice) const {
-  IResearchLinkMeta rhs;
-  std::string errorField;
-
-  return rhs.init(slice, errorField) && _meta == rhs;
-}
-
 void IResearchLinkCoordinator::toVelocyPack(
     arangodb::velocypack::Builder& builder,
     std::underlying_type<arangodb::Index::Serialize>::type flags
 ) const {
-  TRI_ASSERT(_view);
-  TRI_ASSERT(!builder.isOpenObject());
+  if (builder.isOpenObject()) {
```

Comment:
why not just keep the assertion ?

Reply:
an assertion failure is a call to terminate() IIRC, so that'd kill the process